README.md

```diff
@@ -1,2 +1,406 @@
+
+✅ Easy testing: every type has a memberwise initializer
+
+✅ Clear error messages: never debug generated code
+
+## Using SafeDI
+
+SafeDI utilizes Swift Macros and Swift Package Manager plugins to read your code and generate a dependency tree that is validated at compile time. Dependencies can either be instantiated by SafeDI or forwarded into the SafeDI dependency tree.
+
+Opting a type into the SafeDI dependency tree is straightforward: add the `@Instantiable` macro to your type declaration, and decorate your type‘s dependencies with macros that signal the lifecycle of each property. If a type is declared in third-party code, you can opt it into SafeDI by declaring an extension of the type in your code and decorating it with the same `@Instantiable` macro.
+
+There are a total of four macros in the SafeDI library:
+
+| Macro  | Decorating | Usage |
+| ------ | ----------- | ----- |
+| [`@Instantiable`](#instantiable) | Type or extension declaration | Makes a type capable of being instantiated by SafeDI. |
+| [`@Instantiated`](#instantiated) | Property declaration | Instantiates an instance or value when the enclosing `@Instantiable`-decorated type is instantiated. |
+| [`@Forwarded`](#forwarded) | Property declaration | Propagates a runtime-created instance or value (e.g. a User object, network response, or customer input) down the dependency tree. |
+| [`@Received`](#received) | Property declaration | Receives an instance or value from an `@Instantiated` or `@Forwarded` property further up the dependency tree. |
+
+Let‘s walk through each of these macros in detail.
+
+### @Instantiable
+
+Type declarations decorated with [`@Instantiable`](Sources/SafeDI/PropertyDecoration/Instantiable.swift) are able to be instantiated by SafeDI. Types decorated with this macro can instantiate other `@Instantiable` dependencies, forward dependencies injected from outside of SafeDI, or receive dependencies instantiated or forwarded by objects further up the dependency tree.
+
+SafeDI is designed to make instantiating and receiving dependencies _simple_, without requiring engineers to think about abstract dependency injection (DI) concepts. That said, for those familiar with other DI systems: each `@Instantiable` type is its own [Scope](https://medium.com/@aarontharris/scope-dependency-injection-6fc25beffc9c). For those unfamiliar with DI terminology, know that each `@Instantiable` type is responsible for retaining its dependencies, and that every `@Instantiated` or `@Forwarded` dependency is available to all transitive child dependencies.  
+
+Every `@Instantiable`-decorated type must be:
+
+1. `public` or `open`
+
+2. Have a `public init(…)` or `open init(…)` that receives every injectable property
+
+The `@Instantiable` macro guides engineers through satisfying these requirements with build-time FixIts.
+
+Here is a sample `UserService` implementation that is `@Instantiable`:
+
+```swift
+import SafeDI
+
+public final class UserService {
```

Comment:
I think this is missing an `@Instantiable` macro 😂

Reply:
Oh my god. Simplified this example a little too much didn't I.

Reply:
```suggestion
@Instantiable
public final class UserService {
```